sdlib/d/gc/collector.d

```diff
@@ -74,6 +74,12 @@ private:
 		threadCache.flush();
 
 		collect(gcCycle);
+
+		/*
+		 * Removing roots never shrinks the roots array. This must be
+		 * called periodically, and not while doing a collection.
+		 */
```

Comment:
This is missing the why.

Reply:
Updated, let me know if this is clearer. The reason we have to call this is because we can't realloc while actually removing the root. Essentially we cannot realloc inside a finalizer.